tests/suites/test_suite_psa_crypto.function

```diff
@@ -3769,6 +3769,15 @@ void sign_deterministic( int key_type_arg, data_t *key_data,
     ASSERT_COMPARE( output_data->x, output_data->len,
                     signature, signature_length );
 
+#if defined(MBEDTLS_TEST_DEPRECATED)
+    PSA_ASSERT( psa_asymmetric_sign( handle, alg,
```

Comment:
I was thinking about a potential gap in testing deprecated stuff here:
If `psa_asymmetric_sign` somehow happens to return `PSA_SUCCESS` on error, but doesn't modify `signature` in the process, the next `ASSERT_COMPARE` will succeed anyway, since there's old data in the buffer. 
Not sure if it's worth to add buffer zeroization here, since `psa_asymmetric_sign` should not behave badly.

Reply:
I don't think it's useful to test deprecated identifiers as black boxes. My intent was a gray-box test, where we assume that the deprecated identifiers are defined as aliases but they may be defined as the wrong alias, or pass the wrong parameter, or have syntax errors in a macro expansion. Zeroing `signature` here is cheap and useful so I'll do it.